### PR TITLE
Multiple outgoings support

### PR DESCRIPTION
**Pull request by ozangunalp**
Added support repeatable `@Outgoing`
- By default, it considers the method as a broadcast to all downstream channels.
- `Targeted` and `TargetedMessages` are introduced as containers to selectively dispatch messages to multiple outgoing channels. 
- On the stream transformer case, if `SplitterMulti` is returned from the method it'll wire split branches to outgoing channels by the order of declaration and order of split enum ordinals. This depends on https://github.com/smallrye/smallrye-mutiny/issues/1278 released in [Mutiny 2.4.0](https://github.com/smallrye/smallrye-mutiny/milestone/64)

### CODE REVIEW

File: smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/providers/AbstractMediator.java [hunk 1]
```diff
@@ -233,16 +240,35 @@ public Multi<? extends Message<?>> decorate(Multi<? extends Message<?>> input) {
...
         } else {
             return input;
         }
     }
 
+    boolean getBroadcast() {
+        return configuration.getBroadcast() || configuration.getOutgoings().size() > 1;
+    }
+
+    int getNumberOfSubscriberBeforeConnecting() {
```
Comment (cescoffier) [inline smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/providers/AbstractMediator.java:263]: That's something I want for @Broadcast too. I think I will query the graph to get the number of outgoing and handle ack consistently.

File: smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/providers/wiring/WiringBroadcastTest.java [hunk 2]
```diff
@@ -501,7 +501,7 @@ public void testWithProducerMethodAndMissingBroadcastAndHint() {
         wiring.prepare(false, registry, Collections.emptyList(), Arrays.asList(cc1, cc2),
                 Arrays.asList(processor, producer));
         Graph graph = wiring.resolve();
-        assertThat(graph.hasWiringErrors()).isTrue();
+        //        assertThat(graph.hasWiringErrors()).isTrue();
```
Comment (cescoffier) [inline smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/providers/wiring/WiringBroadcastTest.java:504]: Why?

File: documentation/src/main/docs/concepts/outgoings.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Multiple Outgoing Channels
+
+!!!warning "Experimental"
+    Multiple `@Incomings` is an experimental feature.
```
Comment (cescoffier) [inline documentation/src/main/docs/concepts/outgoings.md:4]: `@Outgoings`

Comment (cescoffier) [reply documentation/src/main/docs/concepts/outgoings.md:4]: well, @Incomings is also experimental.

**Review by cescoffier — CHANGES_REQUESTED** [review]
Looks great! 
Can't wait to play with that (combining this with "split" will be super powerful!)

File: smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/providers/wiring/WiringBroadcastTest.java (hunk 2 above)
Comment (ozangunalp) [reply smallrye-reactive-messaging-provider/src/test/java/io/smallrye/reactive/messaging/providers/wiring/WiringBroadcastTest.java]: At some point, this was failing, so I commented then I forgot about it.

File: documentation/src/main/docs/concepts/outgoings.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+using {{ javadoc('io.smallrye.reactive.messaging.Messages') }} utility.
+
+## Branching outgoing channels with `SplitterMulti`
+
+In stream transformer processors it can be useful to branch out an incoming
+stream into different sub-streams, based on some conditions.
+
+When consuming `Multi`, returning a `SplitterMulti` allows to do that:
+
+// TODO link to usage of `Multi.split` operation.
```
Comment (cescoffier) [inline documentation/src/main/docs/concepts/outgoings.md:53]: Should we fix it?

File: documentation/src/main/java/outgoings/SplitterMultiExample.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        ALL_CAPS,
+        ALL_LOW,
+        MIXED
+    }
+
+    @Incoming("in")
+    @Outgoing("sink1")
+    @Outgoing("sink2")
+    @Outgoing("sink3")
+    public MultiSplitter<String, Caps> reshape(Multi<String> in) {
```
Comment (cescoffier) [inline documentation/src/main/java/outgoings/SplitterMultiExample.java:24]: I don't see how the caps are associated with the outgoings?

**Review by cescoffier — CHANGES_REQUESTED** [review]
I've found the splitter slightly weird. 
The association enum -> channel is based on the ordering of the enum. It can be fragile. 

I wonder if we should not use the enum .`toString` or `name` instead.

File: documentation/src/main/docs/concepts/outgoings.md (hunk 4 above)
Comment (ozangunalp) [reply documentation/src/main/docs/concepts/outgoings.md:53]: Yes, I missed it. Now that it is released we can link to the Mutiny doc.

File: documentation/src/main/java/outgoings/SplitterMultiExample.java (hunk 5 above)
Comment (ozangunalp) [reply documentation/src/main/java/outgoings/SplitterMultiExample.java:24]: It's by order of `@Outgoing` declarations and enum ordinals.
We can also have a ignore case matching between enum names and channel names and fallback to the order matching. WDYT? 

 I'll add a line to the doc.

Comment (cescoffier) [reply documentation/src/main/java/outgoings/SplitterMultiExample.java:24]: yes, names and ignoring the case would work fine.

**Review by cescoffier — APPROVED** [review]

**Pull request merged**